### ISSUE TITLE
Update AuthHandler to account for the new AuthorizationProvider

### ISSUE THREAD

**Issue by stephanebastian** [eclipse-vertx/vertx-auth#315]

**Comment by stephanebastian** [comment]
Right now the AuthHandler take care of Authenticating and checking authorizations.
I suggest we add the following interface:
```
AuthorizationHandler {
  AuthorizationHandler setAuthorization(Authorization authorization);
  AuthorizationHandler addAuthorizationProvider(AuthorizationProvider authorizationProvider);
  void authorize(RoutingContext context, Handler<AsyncResult<Void>> handler);
}

```

**Comment by stephanebastian** [comment]
closed in favor of vert-x3/vertx-web#1467
